### DISCUSSION TITLE
`FILESYSTEM` events

### DISCUSSION THREAD

**Discussion by domwhewell-sage** [blacklanternsecurity/bbot#1272] [General]
Don't think there's been a discussion on this yet but I recall we mentioned `FILESYSTEM` events would be really cool!

A folder spider type module could be created that handles `FILESYSTEM` events tagged with `folder` and raises `FILESYSTEM` events tagged with file which could be ingested by `excavate.py` to extract URLs etc.

A potential enhancement for `excavate.py` could be to extract docker run commands and alike and raise them as `CODE_REPOSITORY` events, lets say a README.md contains the command `docker run -it blacklanternsecurity/bbot --help` excavate could then raise this as `https://hub.docker.com/r/blacklanternsecurity/bbot`

**Comment by TheTechromancer** [comment]
I like this! The detection of `CODE_REPOSITORY` from shell commands is a good idea. I've been wanting to make a [`code_repository` module](https://github.com/blacklanternsecurity/bbot/issues/1218) similar to what we currently have for `social`, that basically extracts code repositories from other events like URLs or HTTP_RESPONSE, etc.
